presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkDiskPage.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.util.List;
+
+public class PrestoSparkDiskPage
+        implements Serializable, PrestoSparkTaskOutput
+{
+    private long broadcastId;
+    private final URI filePathURI;
+    private final long uncompressedSizeInBytes;
+    private final long compressedSizeInBytes;
+    private final List<Long> pageChecksum;
```

Comment:
In some corner cases pages could be quite small. Instead of storing a checksum for every page I would recommend storing an overall checksum for the entire file to save memory

Reply:
Have you consider `java.util.zip.CRC32`? It is incremental and blazing fast (from what I know it is hardware accelerated) .